src/Tickets/Commerce/Gateways/Stripe/Payment_Intent.php

```diff
@@ -151,13 +153,27 @@ public static function create( Value $value, $retry = false ) {
 	 */
 	public static function create_from_cart( Cart $cart, $retry = false ) {
 		$items = tribe( Order::class )->prepare_cart_items_for_order( $cart );
-
 		if ( empty( $items ) ) {
 			return [];
 		}
 
 		$value = tribe( Order::class )->get_value_total( array_filter( $items ) );
 
+		/**
+		 * Filters the value and items before creating a Payment Intent.
+		 *
+		 * @since TBD
+		 *
+		 * @param Value $value The total value of the cart.
+		 * @param array $items The items in the cart
+		 */
+		$value = apply_filters( 'tec_tickets_commerce_stripe_create_from_cart', $value, $items );
+
+		// Ensure we have a Value object returned from the filters.
+		if ( ! $value instanceof Value ) {
+			throw new RuntimeException( esc_html__( 'Value object not returned from filter', 'event-tickets' ) );
```

Comment:
Attempting a conversion before failing. I agree that this would be much better even just using a float during the filtering.

I think in general we are overusing objects for very simple calculations in this project but its too late now to change that.

Reply:
FWIW, the idea of the Exception is that a developer using that filter will be immediately alerted if they use it incorrectly. That way it can help prevent hard-to-find errors that can sometimes come with filters.

Reply:
We did the best we could but trying to fix numeric incoming values. A doing it wrong now if every triggered it will also result in a fatal. An exception at least has an explanation message.